### PR TITLE
[New Model]: Business partner certificate 1.0.0

### LINKED ISSUE #415: [New Model]: business partner certificates
## Model Description

A semantic data model for business partner certificates is a structured representation of data that defines the relationships, attributes, and meaning of various elements related to certificates issued to or by business partners. Such a model helps solution providers and data providers to manage and validate certificates.

This model serves as a foundation for database design, data governance, interfaces and application development in the context of certificates and their associated business partners.

**Responsibles for this topic:**
@SujitMBRDI , @maximilianong

## Relevant Standards
Standard comes with the release 24.03.

## Example Data
The data model describes the attributes for business partner certificates.
[Catena-X Semantic Model Certificates.xlsx](https://github.com/eclipse-tractusx/sldt-semantic-models/files/13349139/Catena-X.Semantic.Model.Certificates.xlsx)

## MS1 Criteria

- [ ] The proposed aspect model does not exist already
- [ ] The proposed aspect model does not extend an existing aspect but introduces completely new functionality
- [ ] The proposal references a Catena-X use case
- [ ] Relevant standards are mentioned/linked
- [ ] A first draft of the data structure is provided in an example
- [ ] A modeling steward has been assigned

### PR DESCRIPTION
**Pull request by SujitMBRDI**
## Description
In this pull request, created data model for business partner certificates in structured representation of data that defines the relationships, attributes, and meaning of various elements related to certificates issued to or by business partners.

Closes #415 
Responsibles for this topic:
@maximilianong, @chkrenn

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.5.1)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [x] Property and the referenced Characteristic should not have the same name
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the SAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] generated json schema validates against example json payload
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'
- [x] If a previous model exists, model deprecation has been checked for previous model
- [x] The release date in the Release Note is set to the date of the MS3 approval

### CODE REVIEW

File: io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl [hunk 1]
```diff
@@ -0,0 +1,157 @@
+@prefix samm: <urn:samm:org.eclipse.esmf.samm:meta-model:2.1.0#> .
```
Comment (agg3fe) [inline io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl:1]: please add copyright headers

File: io.catenax.shared.business_partner_certificate/1.0.0/gen/BusinessPartnerCertificate.html [hunk 2]
```diff
@@ -0,0 +1,307 @@
+{
```
Comment (agg3fe) [inline io.catenax.shared.business_partner_certificate/1.0.0/gen/BusinessPartnerCertificate.html:1]: gen folder needs to be removed

File: io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl [hunk 3]
```diff
@@ -0,0 +1,157 @@
+@prefix samm: <urn:samm:org.eclipse.esmf.samm:meta-model:2.1.0#> .
+@prefix samm-c: <urn:samm:org.eclipse.esmf.samm:characteristic:2.1.0#> .
+@prefix samm-e: <urn:samm:org.eclipse.esmf.samm:entity:2.1.0#> .
+@prefix unit: <urn:samm:org.eclipse.esmf.samm:unit:2.1.0#> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix : <urn:samm:io.catenax.business_partner_certificate:1.0.0#> .
+@prefix ext-number: <urn:samm:io.catenax.shared.business_partner_number:1.0.0#> .
```
Comment (agg3fe) [inline io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl:9]: version 2 can be used of bpn dependency.

Comment (agg3fe) [reply io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl:9]: Although you will not see any issues with v1 also, but it's better to use the latest versions. So I always recommend to use latest versions.

File: io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl (hunk 1 above)
Comment (SujitMBRDI) [reply io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl:15]: Added copyright headers

File: io.catenax.shared.business_partner_certificate/1.0.0/gen/BusinessPartnerCertificate.html (hunk 2 above)
Comment (SujitMBRDI) [reply io.catenax.shared.business_partner_certificate/1.0.0/gen/BusinessPartnerCertificate.html]: Removed complete gen folder

File: io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl (hunk 3 above)
Comment (SujitMBRDI) [reply io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl]: Adapted with version 2.0.0 for business partner number

**Comment by SujitMBRDI** [comment]
Hello @agg3fe, I am new to this repo. Kindly confirm about gen folder has to be removed completely(which i have done is recent commit) and do i need to squash the commits as well.

**Comment by agg3fe** [comment]
> Hello @agg3fe, I am new to this repo. Kindly confirm about gen folder has to be removed completely(which i have done is recent commit) and do i need to squash the commits as well.

@SujitMBRDI Thanks for removing gen folder.
No, you don't need to squash the commits here.

File: io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl [hunk 4]
```diff
@@ -0,0 +1,171 @@
+#######################################################################
+# Copyright (c) 2023-2024 Contributors to the Eclipse Foundation
```
Comment (agg3fe) [inline io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl:2]: only 2024 is required.

File: io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+   samm:exampleValue "BPNL00000003AYRE".
+
+
+ :type a samm:Property;
+    samm:preferredName "Type"@en;
+    samm:description "Type of the certificatation as defined on the document like IS09001, IATF 16949 or other"@en;
+    samm:characteristic [ a samm-c:SingleEntity ; samm:dataType :certificateTypeEntity].
+	
+	
+ :certificateTypeEntity a samm:Entity;
```
Comment (agg3fe) [inline io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl:45]: the identifiers for all model elements start with a capital letter except for properties. So `certificateTypeEntity` and other identifiers should start with capital letter.

File: io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+@prefix unit: <urn:samm:org.eclipse.esmf.samm:unit:2.1.0#> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix : <urn:samm:io.catenax.business_partner_certificate:1.0.0#> .
+@prefix ext-number: <urn:samm:io.catenax.shared.business_partner_number:2.0.0#> .
+
+:BusinessPartnerCertificate a samm:Aspect ;
+   samm:preferredName "BusinessPartnerCertificate"@en ;
+   samm:description "A business partner certifcate describes a certification (eg ISO9001, IATF-16949) via a certifcate document  of a Catena-X business partner"@en ;
```
Comment (agg3fe) [inline io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl:27]: typo `certifcate`. please correct here and other places in model also. this typo is in multiple places.

File: io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+:businessPartnerNumber a samm:Property ;
+   samm:preferredName "BusinessPartnerNumber"@en ;
+   samm:description "The BPN of the certified legal entity "@en ;
+   samm:characteristic ext-number:BpnlTrait ;
+   samm:exampleValue "BPNL00000003AYRE".
+
+
+ :type a samm:Property;
+    samm:preferredName "Type"@en;
+    samm:description "Type of the certificatation as defined on the document like IS09001, IATF 16949 or other"@en;
```
Comment (agg3fe) [inline io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl:41]: typo `certificatation`

File: io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+    samm:description "Registration number of the certificate as in the certification document"@en;
+    samm:characteristic samm-c:Text;
+    samm:exampleValue "12 198 54182 TMS".
+
+
+  :areaOfApplication a samm:Property;
+    samm:preferredName "area of application"@en;
+    samm:description "Details on which areas / application types a certification is valid "@en;
+    samm:characteristic samm-c:Text;
+    samm:exampleValue "Development, Marketing und Sales and also Procurement for interiour components".
```
Comment (agg3fe) [inline io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl:73]: typo `interiour`

File: io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+  
+ :enclosedSites a samm:Property;
+    samm:preferredName "Enclosed Sites"@en;
+    samm:description "Optional enclosed sites of the legal entity which are enclosed by the certification "@en;
+	samm:characteristic  [ a samm-c:List ; samm:dataType :enclosedSiteEntity].
+	
+	  
+ :enclosedSiteEntity a samm:Entity;
+    samm:preferredName "Defines enclosed sites "@en;
+    samm:description "Optional enclosed sites where certificatation is valid including details about area of application"@en;
```
Comment (agg3fe) [inline io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl:84]: typo `certificatation`

File: io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+   samm:dataType xsd:date;
+   samm:exampleValue "2026-01-24"^^xsd:date.
+  
+:validator a samm:Property ;
+   samm:preferredName "validator"@en ;
+   samm:description "The BPN of the data service provider who validate the given certificate"@en ;
+   samm:characteristic :trustValidatorCharacteristic.
+
+:trustValidatorCharacteristic a samm:Characteristic;
+	samm:preferredName "validitor Characteristic"@en;
```
Comment (agg3fe) [inline io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl:118]: typo `validitor`

File: io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+	samm:description "The BPN of the data service provider who validate the given certificate"@en ;
+    samm:dataType :trustValidatorEntitity . 
+   
+   
+:trustValidatorEntitity a samm:Entity ;
```
Comment (agg3fe) [inline io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl:123]: typo in property name `trustValidatorEntitity`. please correct this and references.

File: io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+@prefix samm-e: <urn:samm:org.eclipse.esmf.samm:entity:2.1.0#> .
+@prefix unit: <urn:samm:org.eclipse.esmf.samm:unit:2.1.0#> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix : <urn:samm:io.catenax.business_partner_certificate:1.0.0#> .
+@prefix ext-number: <urn:samm:io.catenax.shared.business_partner_number:2.0.0#> .
+
+:BusinessPartnerCertificate a samm:Aspect ;
+   samm:preferredName "BusinessPartnerCertificate"@en ;
```
Comment (agg3fe) [inline io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl:26]: preferredName should be human readable and follow normal orthography (e.g., no camel case but normal word separation).
Please correct here and in other places in the model.

File: io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl (hunk 4 above)
Comment (agg3fe) [reply io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl:2]: you did not add your teams name in copyright headers?

File: io.catenax.shared.business_partner_certificate/RELEASE_NOTES.md [hunk 13]
```diff
@@ -0,0 +1,7 @@
+# Changelog
+
+All notable changes to this model will be documented in this file.
+
+## [1.0.0] 23.02.2024
```
Comment (agg3fe) [inline io.catenax.shared.business_partner_certificate/RELEASE_NOTES.md:5]: you can change the date to 4th March if you need MS3 approval in next weekly meeting.

**Review by agg3fe — CHANGES_REQUESTED** [review]
please check my comments

**Comment by agg3fe** [comment]
@maximilianong I have reviewed the PR and gave some comments. Do you still need tomorrow's meeting?

File: io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl (hunk 4 above)
Comment (SujitMBRDI) [reply io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl]: Added team details asn adapted year as 2024

File: io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl [hunk 14]
```diff
@@ -0,0 +1,171 @@
...
+
+ :type a samm:Property;
+    samm:preferredName "Type"@en;
+    samm:description "Type of the certificatation as defined on the document like IS09001, IATF 16949 or other"@en;
+    samm:characteristic [ a samm-c:SingleEntity ; samm:dataType :certificateTypeEntity].
+	
+	
+ :certificateTypeEntity a samm:Entity;
+    samm:preferredName "Entity certificate Type"@en;
+    samm:description "Detailled entity of the certificate like IS09001:2015, IATF 16949:2015 or other, valid types are registered at BPN metadatacontroller"@en;
```
[the remark this answers is not in the document]
Comment (SujitMBRDI) [reply io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl]: Fixed typo

File: io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl (hunk 8 above)
Comment (SujitMBRDI) [reply io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl]: Fixed typo

File: io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl (hunk 9 above)
Comment (SujitMBRDI) [reply io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl]: Fixed typo

File: io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl (hunk 10 above)
Comment (SujitMBRDI) [reply io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl]: Fixed typo

File: io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl (hunk 11 above)
Comment (SujitMBRDI) [reply io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl]: Fixed typo

File: io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl (hunk 12 above)
Comment (SujitMBRDI) [reply io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl]: Done

File: io.catenax.shared.business_partner_certificate/RELEASE_NOTES.md (hunk 13 above)
Comment (SujitMBRDI) [reply io.catenax.shared.business_partner_certificate/RELEASE_NOTES.md]: Adapted date as 4th March

**Comment by SujitMBRDI** [comment]
@agg3fe, as we discussed yesterday, we have fixed all comments and please have a look on pull request again.
Thanks for your support :)

**Review by agg3fe — APPROVED** [review]
LGTM

File: io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl [hunk 15]
```diff
@@ -0,0 +1,172 @@
...
+  
+ 
+ :uploader a samm:Property ;
+   samm:preferredName "Certifcate uploader"@en ;
+   samm:description "The BPN of the business partner who originally provided the certifcate data or document e.g. Mercedes Benz AG"@en ;
+   samm:characteristic ext-number:BpnlTrait ;
+   samm:exampleValue "BPNL00000003AYRE".
+
+
+ :documentID a samm:Property ;
```
Comment (agg3fe) [inline io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl:167]: @SujitMBRDI do you want to make it a UUID? If yes, then you can use shared model uuid for this.

Comment (SujitMBRDI) [reply io.catenax.shared.business_partner_certificate/1.0.0/BusinessPartnerCertificate.ttl:167]: Hello @agg3fe, 
As per the Christian @chr-krenn, it is not necessary to define a global UUID here from a semantic model perspective

File: io.catenax.shared.business_partner_certificate/RELEASE_NOTES.md [hunk 16]
```diff
@@ -0,0 +1,7 @@
+# Changelog
+
+All notable changes to this model will be documented in this file.
+
+## [1.0.0] 04.03.2024
```
Comment (agg3fe) [inline io.catenax.shared.business_partner_certificate/RELEASE_NOTES.md:5]: change the date to 11th March.

**Review by maximilianong — APPROVED** [review]

File: io.catenax.shared.business_partner_certificate/RELEASE_NOTES.md (hunk 16 above)
Comment (SujitMBRDI) [reply io.catenax.shared.business_partner_certificate/RELEASE_NOTES.md]: > change the date to 11th March.

Hi @agg3fe, I have adapted date to 11th March.

**Pull request merged**